lldb/test/Shell/SymbolFile/DWARF/delayed-definition-die-searching.test

```diff
@@ -0,0 +1,40 @@
+# Test definition DIE searching is delayed until complete type is required.
+
+# RUN: split-file %s %t
+# RUN: %clangxx_host %t/main.cpp %t/t1_def.cpp -g -gsimple-template-names -o %t.out
+# RUN: %lldb -b %t.out -s %t/lldb.cmd | FileCheck %s
+
+# CHECK:      (lldb) p v1
+# CHECK: DWARFASTParserClang::ParseTypeFromDWARF{{.*}}DW_TAG_structure_type (DW_TAG_structure_type) name = 't2'
+# CHECK: DWARFASTParserClang::ParseTypeFromDWARF{{.*}}DW_TAG_structure_type (DW_TAG_structure_type) name = 't1'
+# CHECK: DWARFASTParserClang::ParseTypeFromDWARF{{.*}}DW_TAG_base_type (DW_TAG_base_type) name = 'int'
+# CHECK: DW_TAG_structure_type (DW_TAG_structure_type) 't2' resolving forward declaration...
+# CHECK: (t2<t1<int> >)  {}
+# CHECK: (lldb) p v2
+# CHECK: DWARFASTParserClang::ParseTypeFromDWARF{{.*}}DW_TAG_structure_type (DW_TAG_structure_type) name = 't1'
+# CHECK: DWARFASTParserClang::ParseTypeFromDWARF{{.*}}DW_TAG_base_type (DW_TAG_base_type) name = 'int'
+# CHECK: DW_TAG_structure_type (DW_TAG_structure_type) 't1' resolving forward declaration...
+# CHECK: (t1<int>)  {}
+
+#--- lldb.cmd
```

Comment:
Might be better to test this without `-gsimple-template-names`, because it's a more generally valuable feature than that?

Oh, but that "look for the child count through a pointer" issue comes up and it's hard to reach this situation without `-gsimple-template-names`? (Maybe indirectly - if you print a struct with a pointer - perhaps that doesn't do the child count searching through the pointer?)

Reply:
I tried something like this:
```
struct Bar {
  int x;
  Bar(int x): x(x) {}
};
struct Foo {
  Bar* bar;
  Foo(int x) {
    bar = new Bar(x);
  }
};
```
In another CU, print variable with type `Foo*`, but it still tries to complete both `Foo` and `Bar` because the looking-up  child count through a pointer issue.

Reply:
Hmm - what about template parameters, even without simplified template names?

`t1<t2>` t2 should be a declaration, without looking up the definition DIE in this case? (though perhaps clang/LLDB doesn't do that correctly, in which case maybe `t1<t2*>` might work)

Reply:
Yeah, that would work. I have an example at https://discourse.llvm.org/t/rfc-delay-definition-die-searching-when-parse-a-declaration-die-for-record-type/78526/15.